src/ocean/text/convert/Formatter.d

```diff
@@ -379,9 +379,51 @@ private void handle (T) (T v, FormatInfo f, scope FormatterSink sf, scope ElemSi
     static if (is(T == typeof(null)))
         se("null", f);
 
-    // Cannot print enum member name in D1, so just print the value
-    else static if (is (T V == enum))
-        handle!(V)(v, f, sf, se);
+
+    // Pretty print enum
+    // Note that switch is only available for string and integer based enums.
+    // However, since it expands to a jump table for integers and a binary
+    // search for strings, we still want to special case it.
+    else static if (is(T V == enum) && canSwitchOn!(T))
```

Comment:
minor: don't we already allow "less parenthesis" style like `canSwitchOn!T`?

Reply:
Right, it was just the way the author did it.